File: mahasiswa.py
from human import Human
# kelas Mahasiswa sebagai anak yang diwariskan dari kelas Human
class Mahasiswa(Human):
    # access modifier private
    __nim = "";
    __nama = "";
    __jenis_kelamin = '';
    __fakultas = "";

    # constructor
    def __init__(self,  nama = "", nim = "",jenis_kelamin = '', fakultas = ""):
        self.__nama = nama
        self.__nim = nim
        self.__jenis_kelamin = jenis_kelamin
        self.__fakultas = fakultas
        self.matkul = None #komoditas dari kelas Prodi
        self.univ = None #komoditas dari kelas SivitasAkademik

    # penggunaan getter dan setter
    def get_nim(self):
        return self.__nim
    
    def set_nim(self, nim):
        self.__nim = nim

    def get_nama(self):
        return self.__nama
    
    def set_nama(self, nama):
        self.__nama = nama  

    def get_jenis_kelamin(self):
        return self.__jenis_kelamin
    
    def set_jenis_kelamin(self, jenis_kelamin):
        self.__jenis_kelamin = jenis_kelamin

    def get_fakultas(self):
        return self.__fakultas
    
    def set_fakultas(self, fakultas):
        self.__fakultas = fakultas      

    

File: cpp/mahasiswa.cpp
//import library
#include <bits/stdc++.h>
#include "human.cpp"

//using standard namespace
using namespace std;
//Mahasiswa sebagai anak yang diwariskan dari Human
class Mahasiswa : public Human
{
    //access modifier private
    private:
        string nim;
        string nama;
        string jenis_kelamin;
        string fakultas;
    //access modifier public
    public:
        //constructor
        string matkul;
        string univ;
        Mahasiswa()
        {
            this->nim = " ";
            this->nama = " ";
            this->jenis_kelamin = " ";
            this->fakultas = " ";
            this->matkul =" ";
            this->univ=" ";

        }

        //penggunaan getter dan setter
        string get_nim()
        {
            return this->nim;
        }

        void set_nim(string nim)
        {
            this->nim = nim;
        }

        string get_nama()
        {
            return this->nama;
        }

        void set_nama(string nama)
        {
            this->nama = nama;
        }

        char get_jenis_kelamin()
        {
            return this->jenis_kelamin;
        }

        void set_jenis_kelamin(string jenis_kelamin)
        {
            this->jenis_kelamin = jenis_kelamin;
        }

        string get_fakultas()
        {
            return this->fakultas;
        }

        void set_fakultas(string fakultas)
        {
            this->fakultas = fakultas;
        }

        //destructor
        ~Mahasiswa()
        {

        }

};

